tests/unit/utils/results/stores/test_redis.py

```diff
@@ -34,7 +35,7 @@ def setUp(self):
         self.task_2_id = 'TASK2-95120ce9-470a-419d-a589-d57b14f3f2ca'
         self.backend.mark_as_done(
             task_id=self.task_2_id,
-            result='all good',
+            result=Decimal('22.41'),
```

Comment:
Modifying this unit test to use a more complicated type for `result` exposed an issue with our Redis JSON deserialization, which I later fixed.

Reply:
i have to ask now, is there a test case for verifying the serialization/deserialization of all supported types?
for the record, i am 100% comfortable only supporting string/(non null) byte sequences as these are some of the only large types CFFI can exchange safely.

Reply:
I'm actually not sure where to find "all supported types." I figured that I'd trust Celery with that, and by using its own decoders I should have the same coverage that their codebase does. 

Over time, we may encounter some types that don't work. But since this has a failsafe to just return the original bytestring, we should be able to detect that and add support when needed.